waspc/src/Wasp/SemanticVersion.hs

```diff
@@ -1,41 +1,104 @@
 module Wasp.SemanticVersion
   ( Version (..),
-    VersionBounds (..),
-    isVersionInBounds,
+    Operator (..),
+    Comparator (..),
+    ComparatorSet (..),
+    Range (..),
+    isVersionInRange,
+    rangeFromVersion,
+    rangeFromVersionsIntersection,
   )
 where
 
+import Data.List (intercalate)
+import Data.List.NonEmpty (NonEmpty)
+import qualified Data.List.NonEmpty as NE
 import Numeric.Natural
 import Text.Printf (printf)
 
+-- Implements SemVer (semantic versioning) by following spec from https://github.com/npm/node-semver .
+
 data Version = Version
   { major :: Natural,
     minor :: Natural,
     patch :: Natural
   }
   deriving (Eq, Ord)
 
+-- | We rely on `show` here to produce valid semver representation of version.
```

Comment:
not really here, but in all the other places but here, right? Maybe add a test (if we don't have it yet) to ensure that is the fact?